app/scripts/metamask-controller.js

```diff
@@ -4930,6 +4930,11 @@ export default class MetamaskController extends EventEmitter {
     );
     ///: END:ONLY_INCLUDE_IF
 
+    const isConfirmationRedesignEnabled =
+      process.env.ENABLE_CONFIRMATION_REDESIGN &&
+      this.preferencesController.store.getState().preferences
```

Comment:
As we're checking the preference each time the middleware is used, should this be a function rather than a static property?

Reply:
great catch! Updated [56a0910](https://github.com/MetaMask/metamask-extension/pull/24182/commits/56a091084771a262da76e74a58e18d53569a7aa8)